compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -497,13 +501,69 @@ OMR::ARM64::TreeEvaluator::cloadEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    return commonLoadEvaluator(node, TR::InstOpCode::ldrhimm, cg);
    }
 
+// also handles vloadi
+TR::Register *
+OMR::ARM64::TreeEvaluator::vloadEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   return commonLoadEvaluator(node, TR::InstOpCode::vldrimmq, cg);
+   }
+
 TR::Register *
 OMR::ARM64::TreeEvaluator::awrtbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 	{
 	// TODO:ARM64: Enable TR::TreeEvaluator::awrtbarEvaluator in compiler/aarch64/codegen/TreeEvaluatorTable.hpp when Implemented.
 	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
 	}
 
+TR::Register *
+OMR::ARM64::TreeEvaluator::inlineVectorBinaryOp(TR::Node *node, TR::CodeGenerator *cg, TR::InstOpCode::Mnemonic op)
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Register *lhsReg = NULL, *rhsReg = NULL;
+
+   lhsReg = cg->evaluate(firstChild);
+   rhsReg = cg->evaluate(secondChild);
+
+   TR_ASSERT(lhsReg->getKind() == TR_VRF, "unexpected Register kind\n");
+   TR_ASSERT(rhsReg->getKind() == TR_VRF, "unexpected Register kind\n");
+
+   TR::Register *resReg;
+   resReg = cg->allocateRegister(TR_VRF);
+
+   node->setRegister(resReg);
+   generateTrg1Src2Instruction(cg, op, node, resReg, lhsReg, rhsReg);
+
+   cg->decReferenceCount(firstChild);
+   cg->decReferenceCount(secondChild);
+   return resReg;
+   }
+
+TR::Register *
+OMR::ARM64::TreeEvaluator::vaddEvaluator(TR::Node *node, TR::CodeGenerator *cg)
```

Comment:
I would prefer this function to be in BinaryEvaluator.cpp instead of OMRTreeEvaluator.cpp, although Power defines evaluators for vectors in OMRTreeEvaluator.cpp.